zigzag/opt/loma/memory_allocator.py

```diff
@@ -297,7 +301,30 @@ def find_best_loop_combination(
                         unrolling."""
                     )
                 continue
-            if accesses_comb <= best_accesses:
+            if (accesses_comb <= best_accesses) and self.is_valid_idx_combination_for_mapping_type(
+                current_loop_idxs, mem_ops
+            ):
                 best_accesses = accesses_comb
                 best_loop_idxs = current_loop_idxs
         return best_loop_idxs
+
+    def is_valid_idx_combination_for_mapping_type(self, idx_comb: list[int], mem_ops: list[MemoryOperand]) -> bool:
+        """! Check if the idx_comb is a valid combination for the mapping_type."""
+        if self.mapping_type == TemporalMappingType.UNEVEN:
+            return True
+        elif self.mapping_type == TemporalMappingType.EVEN:
+            return self.is_valid_idx_combination_for_even_mapping_type(idx_comb, mem_ops)
+        else:
+            raise NotImplementedError(f"Mapping type {self.mapping_type} validity checks not implemented.")
+
+    def is_valid_idx_combination_for_even_mapping_type(self, idx_comb: list[int], mem_ops: list[MemoryOperand]):
+        """! Check if the idx_comb is a valid combination for the EVEN mapping type."""
```

Comment:
A bit more of a descriptive comment on what is actually checked here (and the reasoning behind it) might be nice :)